src/libraries/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -103,22 +103,41 @@ public byte[] GetAddressBytes()
             return (byte[])_address.Clone();
         }
 
-        public static PhysicalAddress Parse(string address)
+        public static PhysicalAddress Parse(string address) => Parse(address.AsSpan());
+
+        public static PhysicalAddress Parse(ReadOnlySpan<char> address)
+        {
+            if (!TryParseInternal(address, out PhysicalAddress value))
+                ThrowBadAddressException(address);
+
+            return value;
+
+            static void ThrowBadAddressException(ReadOnlySpan<char> address) =>
+                throw new FormatException(SR.Format(SR.net_bad_mac_address, new string(address)));
+        }
+
+        public static bool TryParse(string address, out PhysicalAddress value) => TryParseInternal(address.AsSpan(), out value);
+
+        public static bool TryParse(ReadOnlySpan<char> address, out PhysicalAddress value) => TryParseInternal(address, out value);
+
+        private static bool TryParseInternal(ReadOnlySpan<char> address, out PhysicalAddress value)
         {
             int validSegmentLength;
             char? delimiter = null;
             byte[] buffer;
+            value = null;
 
             if (address == null)
```

Comment:
`address` is a struct, so it can't be `null`. You'll want to make sure you're preserving the null string vs empty string behavior on the `string` overloads, and that a 0-length span behaves identical to an empty string -- is there a test for each of these scenarios?

Reply:
It was a mistake, you are right. I just fixed null handling and added tests for null and zero-length inputs.